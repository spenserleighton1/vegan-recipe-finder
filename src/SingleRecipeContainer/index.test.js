import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { SingleRecipeContainer, mapStateToProps } from './'

describe('SingleRecipeContainer', () => {
  it('should match the snapshot when it has a props object', () => {
    const recipe = { 
      title: 'some title',
      publisher: 'some publisher',
      ingredients: ['carrot', 'another carrot'],
      linkUrl: 'www.com',
      id: 666
    }
    const wrapper = shallow(<SingleRecipeContainer recipe={ recipe }/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when props object is empty', () => {
    const recipe = {}

    const wrapper = shallow(<SingleRecipeContainer recipe={recipe}/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should call addFavorite when the button is clicked', () => {
    const recipe = { 
      title: 'some title',
      publisher: 'some publisher',
      ingredients: ['carrot', 'another carrot'],
      linkUrl: 'www.com',
      id: 666
    }
    const mockAddFavorite = jest.fn()
    const wrapper = shallow(<SingleRecipeContainer recipe={ recipe } addFavorite={ mockAddFavorite } />)


    wrapper.find('.favorite-btn').simulate('click')

    expect(mockAddFavorite).toHaveBeenCalled()
  })

  describe('mapStateToProps', () => {
    it('should return a props object', () => {
      const mockState = { singleRecipe: 'sure'};
      const expected = { recipe: 'sure'};

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })
})