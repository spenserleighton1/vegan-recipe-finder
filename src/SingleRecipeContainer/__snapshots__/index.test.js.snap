// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRecipeContainer should match the snapshot when it has a props object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleRecipeContainer
    recipe={
        Object {
            "id": 666,
            "ingredients": Array [
              "carrot",
              "another carrot",
            ],
            "linkUrl": "www.com",
            "publisher": "some publisher",
            "title": "some title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          some title
</h2>,
        <p>
          some publisher
</p>,
        <ul>
          <li>
                    carrot
          </li>
          <li>
                    another carrot
          </li>
</ul>,
        <button
          className="favorite-btn"
          onClick={[Function]}
>
          fav
</button>,
        <a
          href="www.com"
>
          Directions
</a>,
      ],
      "className": "single-recipe",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some title",
        },
        "ref": null,
        "rendered": "some title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some publisher",
        },
        "ref": null,
        "rendered": "some publisher",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              carrot
</li>,
            <li>
              another carrot
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "carrot",
            },
            "ref": null,
            "rendered": "carrot",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "another carrot",
            },
            "ref": null,
            "rendered": "another carrot",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "fav",
          "className": "favorite-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "fav",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Directions",
          "href": "www.com",
        },
        "ref": null,
        "rendered": "Directions",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            some title
</h2>,
          <p>
            some publisher
</p>,
          <ul>
            <li>
                        carrot
            </li>
            <li>
                        another carrot
            </li>
</ul>,
          <button
            className="favorite-btn"
            onClick={[Function]}
>
            fav
</button>,
          <a
            href="www.com"
>
            Directions
</a>,
        ],
        "className": "single-recipe",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some title",
          },
          "ref": null,
          "rendered": "some title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some publisher",
          },
          "ref": null,
          "rendered": "some publisher",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                carrot
</li>,
              <li>
                another carrot
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "carrot",
              },
              "ref": null,
              "rendered": "carrot",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "another carrot",
              },
              "ref": null,
              "rendered": "another carrot",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "fav",
            "className": "favorite-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "fav",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Directions",
            "href": "www.com",
          },
          "ref": null,
          "rendered": "Directions",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SingleRecipeContainer should match the snapshot when props object is empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleRecipeContainer
    recipe={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
